.cicd/submodule-regression-check.sh

```diff
@@ -6,20 +6,25 @@ declare -A BASE_MAP
 if ${TRAVIS:-false}; then
     BASE_BRANCH=$TRAVIS_BRANCH
     CURRENT_BRANCH=${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH} # We default to TRAVIS_BRANCH if it's not a PR so it passes on non PR runs
+    [[ ! -z $TRAVIS_PULL_REQUEST_SLUG ]] && CURRENT_BRANCH=$TRAVIS_COMMIT # Support git log & echo output
 else
     BASE_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH:-$BUILDKITE_BRANCH}
     CURRENT_BRANCH=$BUILDKITE_BRANCH
 fi
+[[ $BASE_BRANCH == $CURRENT_BRANCH ]] && echo 'BASE_BRANCH and CURRENT_BRANCH are the same' && exit 0
```

Comment:
I think this message is misleading because if no pull request information is available (say because the Travis or Buildkite build is run manually on a commit), then `$BASE_BRANCH == $CURRENT_BRANCH` even though there is no concept of a base branch available (which only makes sense within a PR).

I think it makes sense to make this check, print this message, and exit if the base branch and current branch are the same, but only if there actually is a base branch. In the case when no base branch is available, the script should exit earlier (still successfully) but with a different message. For example, when `$TRAVIS` is true, if there is no `$TRAVIS_PULL_REQUEST_BRANCH` then the script should exit with an error message like "No pull request detected. Skipping submodule regression check.". Same goes when `$TRAVIS` is false (i.e. `$BUILDKITE` is true), and there is no `$BUILDKITE_PULL_REQUEST_BASE_BRANCH`.